### DISCUSSION TITLE
Getter with params not reactive?

### DISCUSSION THREAD

**Discussion by 3zzy** [vuejs/pinia#536] [Help and Questions] [answered]
I have this getter:
```
getDataById: (state) => ({ dataId, dataType } = {}) => {
    return state?.[dataType].find(dataObj => dataObj.id === dataId);
},
```
which when used in a component like so:
```
const data = DataStore.getDataById({ dataType: 'dynamic', dataId: props.dataId });
```
... stays `undefined`. Verified in `setupState.data`.
Even tried watching the `data`, not no change detected.

However, if I do the following in console:
```
$vm.setupState.DataStore.getDataById({ dataType: 'dynamic', dataId: 'contributions' })`
```
returns the correct data.

What am I doing wrong here?

Thanks!

**Comment by posva** [comment, marked as answer]
You need a computed where you call the function:

```js
const data = computed(() => DataStore.getDataById({ dataType: 'dynamic', dataId: props.dataId }))

```

**Reply by 3zzy** [reply]
Oh ok, I though getter was already reactive so using computed might cause an issue. 

Many thanks for the quick response and of course the wonderful library.

**Reply by 3zzy** [reply]
@posva Sorry, can I also ask if I wanted to watch the changes, this works:

```
watch(data, (data, prevData) => {
   console.log({data, prevData})
});
```

but just wanted to confirm if thats the best way to do it? I don't know because watching a computed prop doesn't feel right.

**Reply by posva** [reply]
The getter is a computed but you are returning a function and then calling it which returns a non reactive result (same in Vuex)
watching a computed is fine
